### PR TITLE
[LLD] [COFF] Rewrite handling of the /debug: option. NFC.

### PR DESCRIPTION
**Pull request by mstorsjo**
Don't treat the options as unique enum items, but more as flags that can be composed, like the /opt: options.

This still only processes the last option on the command line though, so the behaviour should still remain exactly as it was, in all corner cases.

### CODE REVIEW

**Comment by mstorsjo** [comment]
This goes on top of #75172; only the latter commit should be reviewed in this PR. The overall direction of the larger patchset that this belongs to, was presented/discussed in https://github.com/llvm/llvm-project/pull/74820.

File: lld/COFF/Driver.cpp [hunk 1]
```diff
@@ -1647,13 +1607,47 @@ void LinkerDriver::linkerMain(ArrayRef<const char *> argsArr) {
...
+    std::string s;
+    if (arg->getOption().getID() == OPT_debug)
+      s = "full";
+    else
+      s = StringRef(arg->getValue()).lower();
+    if (s == "fastlink") {
+      warn("/debug:fastlink unsupported; using /debug:full");
+      s = "full";
+    }
+    if (s == "none") {
```
Comment (ZequanWu) [inline lld/COFF/Driver.cpp:1629]: Maybe make `doGC` default to false and set to true here. So the following `doGC = false` can be removed.

Comment (mstorsjo) [reply lld/COFF/Driver.cpp:1629]: But if there's no `/debug:` option at all, then we'd go with the initial value of `doGC` - which should be `true` for that case.

Comment (mstorsjo) [reply lld/COFF/Driver.cpp:1629]: @ZequanWu Can you follow up here?

Comment (ZequanWu) [reply lld/COFF/Driver.cpp:1629]: Oh, I missed that. That makes sense.

**Review by ZequanWu — APPROVED** [review]

**Pull request merged**